src/dosext/mfs/mfs.c

```diff
@@ -1680,6 +1681,56 @@ dos_fs_dev(struct vm86_regs *state)
     return TRUE;
   }
 
+  /* Redirect all fatfs disks */
+  if (WORD(state->ebx) == DOS_SUBHELPER_MFS_REDIR_MAP_ALL_DRV) {
+#if DOSEMU
+    int i, j, ret=0;
+    char s[255];
+
+    Debug0((dbg_fd, "Redirect fatfs drives\n"));
+
+    for (i = 0; i < MAX_HDISKS; i++) {
+      if (hdisktab[i].type == DIR_TYPE && hdisktab[i].fatfs) {
+        snprintf(s, sizeof(s), LINUX_RESOURCE "%s", hdisktab[i].dev_name);
+        j = RedirectDisk(i + 2, s, hdisktab[i].rdonly);
+        Debug0((dbg_fd, "Redirect %c: %s (err = %d)\n", 'C' + i, j ? "failed" : "ok", j));
+
+        ret += j;
+      }
+    }
+
+    SETWORD(&(state->eax), ret == 0);
+    return TRUE;
+#else
+    return FALSE;
+#endif
+  }
+
+  /* Redirect all printers */
+  if (WORD(state->ebx) == DOS_SUBHELPER_MFS_REDIR_MAP_ALL_PRN) {
```

Comment:
I think so.
If emufs does not have any knowledge
about what it actually redirects (it does
not enumerate drives and printers in your
code), then I think it should better call a
single init. If it will start to enumerate them
and feed to 5f03, then it will keep calling
the same init, but it will just became less bloated.